pvlib/tests/spectrum/test_irradiance.py

```diff
@@ -72,3 +72,73 @@ def test_get_reference_spectra_invalid_reference():
     # test that an invalid reference identifier raises a ValueError
     with pytest.raises(ValueError, match="Invalid standard identifier"):
         spectrum.get_reference_spectra(standard="invalid")
+
+
+def test_average_photon_energy_series():
+    # test that the APE is calculated correctly with single spectrum
+    # series input
+
+    spectra = spectrum.get_reference_spectra()
+    spectra = spectra['global']
+
+    ape = spectrum.average_photon_energy(spectra)
+
+    expected = 1.45017
+
+    assert_allclose(ape, expected, rtol=1e-4)
+
+
+def test_average_photon_energy_dataframe():
+    # test that the APE is calculated correctly with multiple spectra
+    # dataframe input
+
+    spectra = spectrum.get_reference_spectra().T
+    ape = spectrum.average_photon_energy(spectra)
+    expected = [1.36848, 1.45017, 1.40885]
+    assert_allclose(ape, expected, rtol=1e-4)
```

Comment:
If we change the output to be Series when input is DataFrame, then we can verify that in the tests using `assert_series_equal`.  It will check that the output is a Series with matching index.

Reply:
oh I updated the tests by adding `assert isinstance(ape, pd.Series)` into an existing test that uses a dataframe as an input, then noticed this comment. What's the difference...? 
Thanks for the review btw 👍🏾

Reply:
Here is an illustration of the difference:

```python
import pandas as pd
import numpy as np

from numpy.testing import assert_allclose
from pandas.testing import assert_series_equal

times = pd.date_range("2019-01-01", freq="h", periods=3)
s1 = pd.Series([5, 10, 20], index=times)
s2 = pd.Series([5, 10, 20])  # default index is [0, 1, 2]
s3 = pd.Series([5., 10., 20.], index=times)  # dtype=float
a1 = np.array([5, 10, 20])

assert_allclose(s1, s2)  # passes, despite index being different
assert_allclose(s1, s3)  # passes, despite dtype being different
assert_allclose(s1, a1)  # passes, despite type being different

assert_series_equal(s1, s2)  # AssertionError: Series.index are different
assert_series_equal(s1, s3)  # AssertionError: Attribute "dtype" are different: [left]:  int64, [right]: float64
assert_series_equal(s1, a1)  # AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'numpy.ndarray'> instead
```

Reply:
Okay, understood. Please double check the [latest commit](https://github.com/pvlib/pvlib-python/pull/2140/commits/ace5a651496917145fc5010dc711e9a01c6e0e08) that addresses this comment